mvlearn/cluster/mv_coreg_spectral.py

```diff
@@ -0,0 +1,290 @@
+        for the KMeans clustering step.
+
+    objective_ : array-like, shape (n_views, n_iterations)
+        The value of the spectral clustering objective for each view at
+        the end of each iteration.
+
+    Notes
+    -----
+
+    In standard spectral clustering, the eigenvector matrix U for a given view
+    is the new data representation to be used for the subsequent k-means
+    clustering stage. In this algorithm, the objective function has been
+    altered to encourage the pairwise similarities of examples under the new
+    representation to be similar across all views.
+
+    The modified spectral clustering objective for the case of two views is
+    shown and derived in [#3Clu]. In the clustering objective, the
+    hyperparameter lambda trades-off the spectral clustering objectives and
+    the disagreement term.
+
+    For a fixed lambda and n, the objective function is bounded from above and
+    non-decreasing. As such, the algorithm is guaranteed to converge.
+
+    References
+    ----------
+    .. [#3Clu] Kumar A, Rai P, Daumé H (2011) Co-regularized multi-view
+            spectral clustering. Adv Neural Inform Process Syst 24:1413–1421
+
+
+    Examples
+    --------
+    >>> from mvlearn.datasets import load_UCImultifeature
+    >>> from mvlearn.cluster import MultiviewCoRegSpectralClustering
+    >>> from sklearn.metrics import normalized_mutual_info_score as nmi_score
+    >>> # Get 5-class data
+    >>> data, labels = load_UCImultifeature(select_labeled = list(range(5)))
+    >>> mv_data = data[:2]  # first 2 views only
+    >>> mv_spectral = MultiviewCoRegSpectralClustering(n_clusters=5,
+    ...     random_state=10, n_init=100)
+    >>> mv_clusters = mv_spectral.fit_predict(mv_data)
+    >>> nmi = nmi_score(labels, mv_clusters, average_method='arithmetic')
+    >>> print('{0:.3f}'.format(nmi))
+    0.663
+    '''
+
+    def __init__(self, n_clusters=2, v_lambda=2, random_state=None,
+                 info_view=None, max_iter=10, n_init=10, affinity='rbf',
+                 gamma=None, n_neighbors=10):
+
+        super().__init__(n_clusters=n_clusters, random_state=random_state,
+                         info_view=info_view, max_iter=max_iter,
+                         n_init=n_init, affinity=affinity, gamma=gamma,
+                         n_neighbors=n_neighbors)
+
+        self.v_lambda = v_lambda
+        self.objective_ = None
+
+    def _init_umat(self, X):
+
+        r'''
+        Computes the top several eigenvectors of the
+        normalized graph laplacian of a given similarity matrix.
+        The number of eigenvectors returned is equal to n_clusters.
+
+        Parameters
+        ----------
+        X : array-like, shape(n_samples, n_samples)
+            The similarity matrix for the data in a single view.
+
+        Returns
+        -------
+        la_eigs : array-like, shape(n_samples, n_clusters)
+            The top n_cluster eigenvectors of the normalized graph
```

Comment:
it seems like this function actually returns a 3-item tuple, right?

Reply:
Oh, I forgot to change the documentation. I will take care of this.